Add external link icon to Jetpack reader feed link

pfefferle opened the pull request:
Updated the Jetpack integration to add the 'activitypub-external' class to the reader feed link, and added CSS to display a dashicon after external links. This visually indicates external links in the admin interface.

<img width="817" height="327" alt="Screenshot 2025-09-30 at 13 30 44" src="https://github.com/user-attachments/assets/a3f8546f-6c8b-42f6-8f18-366e7269f50f" />

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

<!-- Add a changelog message here -->

</details>

pfefferle committed:
Add external link icon to Jetpack reader feed link

Updated the Jetpack integration to add the 'activitypub-external' class to the reader feed link, and added CSS to display a dashicon after external links. This visually indicates external links in the admin interface.

pfefferle committed:
Update Jetpack connection check in integration

Replaces the use of Automattic\Jetpack\Connection\Manager for checking user connection with Jetpack::is_connection_ready(). This simplifies the connection check and removes an unnecessary use statement.

jeherve commented on integration/class-jetpack.php:
```diff
@@ -104,7 +104,7 @@ public static function add_reader_link( $actions, $item ) {
 			array(
 				'reader' => sprintf(
-					'<a href="%s" target="_blank">%s</a>',
+					'<a href="%s" target="_blank" class="activitypub-external">%s</a>',
```
You may be able to address the a11y issues by following the markup used in core:
https://github.com/WordPress/wordpress-develop/blob/6b6d2d1975cf72ace0f37ba440d380e16bdbb0d2/src/wp-admin/includes/dashboard.php#L1313

As an added bonus, if you use the `dashicons dashicons-external` classes you may not need to add any custom CSS, and you could leave Core handle everything.

pfefferle replied:
It looks awful with the span :(

pfefferle replied:
But if it is the WordPress way ;)

It is at least now consistent with the Site-Health.

<img width="817" height="327" alt="Screenshot 2025-09-30 at 13 30 44" src="https://github.com/user-attachments/assets/a3f8546f-6c8b-42f6-8f18-366e7269f50f" />

jeherve replied:
> It looks awful with the span :(

The icon alone looks terrible imo. :) I can't wait until it gets replaced with the "North East Arrow" character like in Gutenberg screens.

pfefferle committed:
Add aria-label to Reader link for accessibility

The Reader link in the Jetpack integration now includes an aria-label attribute to improve accessibility for screen readers.

pfefferle committed:
Update external link icon and accessibility in Jetpack integration

Removes the CSS-based external link icon and updates the Jetpack integration to use a Dashicons icon within the link markup. Improves accessibility by adding screen reader text indicating the link opens in a new tab.

obenland commented on integration/class-jetpack.php:
```diff
@@ -104,9 +104,11 @@ public static function add_reader_link( $actions, $item ) {
 			array(
 				'reader' => sprintf(
-					'<a href="%s" target="_blank">%s</a>',
+					'<a href="%1$s" target="_blank">%2$s<span class="screen-reader-text"> %3$s</span><span aria-hidden="true" class="dashicons dashicons-external"></span></a>',
```
```suggestion
					'<a href="%1$s" target="_blank">%2$s<span class="screen-reader-text"> %3$s</span><span aria-hidden="true" class="dashicons dashicons-external" style="font-size:1em;line-height:1.7;"></span></a>',
```
Let's make it a bit easier on the eyes:

<img width="482" height="168" alt="Image" src="https://github.com/user-attachments/assets/3860c15d-66d6-43a2-84b6-e9c4d962759b" />

pfefferle replied:
https://github.com/Automattic/wordpress-activitypub/pull/2247#discussion_r2391007255

obenland replied:
Not sure what that's trying to say? My suggestion doesn't change the markup or affect a11y, as far as I'm aware.

If it's about not using custom CSS, in this case it serves a purpose.

pfefferle replied:
Consistency!?

<img width="555" height="250" alt="Screenshot 2025-09-30 at 15 59 55" src="https://github.com/user-attachments/assets/c77b0f84-30ec-47c4-badd-26d0ef0e5bab" />

I added some CSS. I prefer that over inline CSS.

pfefferle committed:
Adjust dashicon styles in followings table

Set font size and line height for dashicons within the followings table to improve visual consistency.

pfefferle committed:
Merge branch 'trunk' into add/external-icon

pfefferle committed:
Merge branch 'trunk' into add/external-icon

pfefferle committed:
Merge branch 'trunk' into add/external-icon
